### PR TITLE
Combobox input ref gadget hacks

### PR DESCRIPTION
**Pull request by nick-mcdonald**
I've got these changes in the `gadget-inc/baseweb/combobox-inputRef` branch that I will submit for upstream review.

In the meantime we can use this for our own fork

### CODE REVIEW

**Comment by airhorns** [comment]
Looks right to me! I imagine the baseui folks may have some feedback but 🚢

**Review by airhorns — APPROVED** [review]

File: documentation-site/components/yard/config/combobox.ts [hunk 1]
```diff
@@ -100,6 +100,11 @@ const ComboboxConfig: TConfig = {
       value: undefined,
       type: PropTypes.String,
       description: 'Name attribute.',
+    },
+    inputRef: {
```
Comment (thegedge) [inline documentation-site/components/yard/config/combobox.ts:104]: I was able to provide a ref for the datetime component:

https://github.com/gadget-inc/gadget/blob/main/packages/web/src/components/render/views/form/DateTimePicker.tsx#L30

One of the issues is that the input is nested, so you have to go:

```js
Input: { props: { overrides: { Input: { props: { ref: forwardProps.ref } } } } },
```

Just wanted to double check to see if that would work here.

Comment (nick-mcdonald) [reply documentation-site/components/yard/config/combobox.ts:104]: https://github.com/gadget-inc/baseweb/pull/2/files#diff-b31b4cdb8a894111f734f0926cc0bef16c4a9e21633a98629f35288b5840014aR307

Unfortunately overriding it means that the internal ref will never get set and that breaks its behaviour. The internal component needs the ref as well as forwarding it along

Comment (thegedge) [reply documentation-site/components/yard/config/combobox.ts:104]: Ahhh, gotcha. That's a whole lot of sad pandas.

**Review by thegedge — APPROVED** [review]

**Pull request merged**